### PR TITLE
[TII][X86] Do not schedule frame-setup/frame-destory instructions

### PR DESCRIPTION
**Pull request by HaohaiWen**
frame-setup/frame-destroy instruction can not be scheduled around by
PostRAScheduler. Their order is critical for SEH.

### CODE REVIEW

**Comment by HaohaiWen** [comment]
Test case is related to schedule model is not easy to create.

File: llvm/lib/Target/X86/X86InstrInfo.cpp [hunk 1]
```diff
@@ -8832,6 +8832,11 @@ bool X86InstrInfo::isSchedulingBoundary(const MachineInstr &MI,
       Opcode == X86::PLDTILECFGV)
     return true;
 
+  // Frame setup and destory can't be scheduled around.
+  if (MI.getFlag(MachineInstr::FrameSetup) ||
+      MI.getFlag(MachineInstr::FrameDestroy))
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrInfo.cpp:8837]: Should we put it in TargetInstrInfo::isSchedulingBoundary?

Comment (HaohaiWen) [reply llvm/lib/Target/X86/X86InstrInfo.cpp:8837]: Lots of arm tests would fail.

Comment (phoebewang) [reply llvm/lib/Target/X86/X86InstrInfo.cpp:8837]: If arm accepts scheduling with frame setup and destory, why can't for X86?

Comment (HaohaiWen) [reply llvm/lib/Target/X86/X86InstrInfo.cpp:8837]: Failed arm tests: https://buildkite.com/llvm-project/github-pull-requests/builds/75530#01904eb7-13d0-451e-b1da-bcb86cd44b60

```
.seh_savexmm %xmm15, 544
.seh_endprologue
```
For X86, most of frame-setup instruction is related to SEH exception restore.
Changing part of its order does not matter. e.g. two independent register save instructions.
Part of them does matter. e.g. .seh_pushreg, moving .seh_endprologue which may affect prologue size. 

Anyway, rescheduling frame-setup instruction does not look like profitable for X86.

Comment (phoebewang) [reply llvm/lib/Target/X86/X86InstrInfo.cpp:8837]: I can't see the reason from the log, but I'm good with this change as long as no performance regression for X86.

**Review by phoebewang — APPROVED** [review]
LGTM.

**Review by KanRobert — APPROVED** [review]
LGTM

**Comment by HaohaiWen** [comment]
Thanks

**Pull request merged**